addons/account/migrations/13.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,244 @@
+            openupgrade.logged_query(
+                env.cr,
+                "ALTER TABLE %(table)s "
+                "ALTER COLUMN %(field)s "
+                "TYPE varchar" % {
+                    'table': table,
+                    'field': field,
+                })
+
+
+def create_account_move_new_columns(env):
+    """Faster way, avoid compute"""
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE account_move
+        ADD COLUMN amount_total_signed numeric""",
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE account_move
+        ADD COLUMN amount_untaxed numeric""",
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE account_move
+        ADD COLUMN amount_untaxed_signed numeric""",
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE account_move
+        ADD COLUMN amount_tax numeric""",
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE account_move
+        ADD COLUMN amount_tax_signed numeric""",
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE account_move
+        ADD COLUMN amount_residual_signed numeric""",
+    )
+
+
+def fill_account_move_line_parent_state(env):
+    """Faster way"""
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE account_move_line
+        ADD COLUMN parent_state varchar""",
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE account_move_line aml
+        SET parent_state = am.state
+        FROM account_move am
+        WHERE am.id = aml.move_id""",
+    )
+
+
+def fill_account_move_line_account_internal_type(env):
+    """Faster way"""
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE account_move_line
+        ADD COLUMN account_internal_type varchar""",
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE account_move_line aml
+        SET account_internal_type = aat.type
+        FROM account_account aa
+        JOIN account_account_type aat ON aa.user_type_id = aat.id
+        WHERE aml.account_id = aa.id""",
+    )
+
+
+def create_res_partner_ranks(env):
+    """Faster way"""
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE res_partner
+        ADD COLUMN customer_rank integer
+        DEFAULT 0""",
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE res_partner ALTER COLUMN customer_rank DROP DEFAULT""",
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE res_partner
+        ADD COLUMN supplier_rank integer
+        DEFAULT 0""",
+    )
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE res_partner ALTER COLUMN supplier_rank DROP DEFAULT""",
+    )
+
+
+def add_helper_invoice_move_rel(env):
+    openupgrade.logged_query(
+        env.cr, """
+        ALTER TABLE account_move
+        ADD COLUMN old_invoice_id integer""",
+    )
```

Comment:
I would also consider creating an index here. This column is used in a few queries as the join condition and will usually be a relatively large table.

Reply:
will check

Reply:
done